marshal.go

```diff
@@ -976,24 +980,31 @@ func (x *GoSNMP) unmarshalHeader(packet []byte, response *SnmpPacket) (int, erro
 }
 
 func (x *GoSNMP) unmarshalPayload(packet []byte, cursor int, response *SnmpPacket) error {
-	var err error
+	if packet == nil || (bytes.Equal(packet, []byte{}) && cursor == 0) {
```

Comment:
```suggestion
	if len(packet) == 0 {
```

Maybe checking `len(packet) == 0` could be enough ? since both nil and empty list will lead to `0`

Any reason to check `cursor == 0` too ? It seems that `requestType := PDUType(packet[cursor])` will fail in any case if packet is empty/nil.

Reply:
Thanks for checking the PR Alexandre. 
Ran some tests and you are right, we can reduce the packet validation to ``len(packet) == 0`` here.
Will update the PR accordingly.